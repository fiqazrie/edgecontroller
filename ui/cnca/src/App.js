/*
 * SPDX-License-Identifier: Apache-2.0
 * Copyright (c) 2019-2020 Intel Corporation
 */

import React, { Component } from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';
import { BrowserRouter as Router } from "react-router-dom";
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Subscriptions from './components/Subscriptions';
import Subscription from './components/Subscription';
import SubscriptionModify from './components/SubscriptionModify';
import Services from './components/Services.js';
import Service from './components/Service.js';
import PacketFlowDescriptors from './components/PacketFlowDescriptors.js';
import PacketFlowDescriptor from './components/PacketFlowDescriptor.js';
import PacketFlowDescriptorAppModify from './components/PacketFlowDescriptorAppModify.js';
import Header from './components/Header';
import ErrorBoundary from './components/ErrorBoundary';

const useStyles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
  },
  header: {
    padding: 20,
    flexGrow: 1,
  },
  main: {
    width: 'auto',
    marginTop: theme.spacing(14),
    marginBottom: theme.spacing(2),
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(4))]: {
      width: 800,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
});

class App extends Component {
  render() {
    const { classes } = this.props;

    return (
      <Router>
        <div>
          <header>
            <Header className={classes.header} />
          </header>
          <main className={classes.main}>
            <ErrorBoundary>
              <Switch>
                <Route
                  exact
                  path="/"
                  render={() => <Redirect to="/services" />}
                />
                <Route
                  exact
                  path="/services"
                  component={Services}
                />
                <Route
                  exact
                  path="/services/create"
                  render={(props) => <Service {...props} createMode={true}/>}
                />
                <Route
                  exact
                  path="/services/:id"
                  component={Service}
                />
                <Route
                  exact
                  path="/subscriptions"
                  component={Subscriptions}
                />
                <Route
                  exact
                  path="/subscriptions/create"
                  render={(props) => <Subscription {...props} createMode={true} />}
                />
                <Route
                  exact
                  path="/subscriptions/edit/:id"
                  component={Subscription}
                />
                <Route
                  exact
                  path="/subscriptions/patch/:id"
                  component={SubscriptionModify}
                />
                <Route
                  exact
                  path="/pfd"
                  component={PacketFlowDescriptors}
                />
                <Route
                  exact
                  path="/pfd/transactions/create"
                  render={(props) => <PacketFlowDescriptor {...props} createMode={true} />}
                />

                 <Route
                  exact
                  path="/pfd/transactions/:tId"
                  component={PacketFlowDescriptor}
                />
                <Route
                  exact
                  path="/pfd/transactions/:tId/applications/:appId"
                  render={(props) => <PacketFlowDescriptorAppModify {...props} />}
                />
 
 
                <Route
                  render={() => <span>404 Not Found</span>}
                />
              </Switch>
            </ErrorBoundary>
          </main>
        </div>
      </Router>
    )
  }
}

App.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(useStyles)(App);
