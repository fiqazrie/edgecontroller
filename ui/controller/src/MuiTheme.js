// SPDX-License-Identifier: Apache-2.0
// Copyright (c) 2019 Intel Corporation

import {createMuiTheme} from "@material-ui/core";
import {blue, indigo} from "@material-ui/core/colors";

const theme = createMuiTheme({
  palette: {
    secondary: {
      main: blue[900]
    },
    primary: {
      main: indigo[700]
    },
    grey: {
      '100': '#F8F8F8',
    }
  },
  typography: {
    useNextVariants: true,
    // Use the system font instead of the default Roboto font.
    fontFamily: [
      '"Lato"',
      'sans-serif'
    ].join(',')
  }
});

export default theme
