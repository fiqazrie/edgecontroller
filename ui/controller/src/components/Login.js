// SPDX-License-Identifier: Apache-2.0
// Copyright (c) 2019-2020 Intel Corporation

import React, { Component } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import { withRouter } from 'react-router-dom'
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import Avatar from '@material-ui/core/Avatar';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Auth from './Auth';

const numeral = require('numeral');

numeral.defaultFormat('0');
const styles = theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: theme.palette.secondary['A100'],
    overflow: 'hidden',
    backgroundSize: 'cover',
    backgroundPosition: '0 400px',
    marginTop: 10,
    padding: 20,
    paddingBottom: 500
  },
  grid: {
    margin: `0 ${theme.spacing(2)}px`
  },
  smallContainer: {
    width: '60%'
  },
  bigContainer: {
    width: '80%'
  },
  stepContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  stepGrid: {
    width: '80%'
  },
  buttonBar: {
    marginTop: 10,
    display: 'flex',
    justifyContent: 'center'
  },
  button: {
    backgroundColor: theme.palette.primary['A100']
  },
  paper: {
    padding: theme.spacing(3),
    textAlign: 'left',
    color: theme.palette.text.secondary
  },
  topInfo: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 42
  },
  formControl: {
    width: '100%'
  },
  loginBox: {
    minHeight: "5rem",
  },
});

class LoginForm extends Component {
  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleLogin = event => {
    event.preventDefault();

    const { username, password } = this.state;

    Auth.login(username, password)
      .then(({ success, errorText }) => {
        if (success) {
          this.props.history.push('/');
          return;
        }

        this.setState({ loginError: true, helperText: errorText })
      });
  };


  constructor(props) {
    super(props);
    this.state = {
      loginError: false,
      helperText: ""
    };

    // This binding is necessary to make `this` work in the callback
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleLogin = this.handleLogin.bind(this);
  }

  render() {
    const { classes } = this.props;

    return (
      <React.Fragment>
        <CssBaseline />
        <div className={classes.root}>
          <Grid container justify="center">
            <Grid spacing={24} alignItems="center" justify="center" container className={classes.grid}>
              <Grid item xs={12} lg={6}>
                <div className={classes.stepContainer}>
                  <div className={classes.smallContainer}>
                    <main className={classes.main}>
                      <CssBaseline />
                      <Paper className={classes.paper}>
                        <Avatar className={classes.avatar}>
                          <LockOutlinedIcon />
                        </Avatar>
                        <Typography component="h1" variant="h5">
                          Controller Login
                        </Typography>
                        <form onSubmit={this.handleLogin} className={classes.form} autoComplete="off">
                          <FormControl margin="normal" required fullWidth>
                            <InputLabel htmlFor="username">Username</InputLabel>
                            <Input error={this.state.loginError} id="username" name="username" autoComplete="off" onChange={this.handleInputChange} autoFocus />
                          </FormControl>
                          <FormControl className={classes.loginBox} margin="normal" required fullWidth>
                            <InputLabel htmlFor="password">Password</InputLabel>
                            <Input error={this.state.loginError} aria-describedby="component-error-text" name="password" type="password" id="password" onChange={this.handleInputChange} autoComplete="off" />

                            {this.state.helperText !== "" ?
                              <FormHelperText id="component-error-text">
                                {this.state.helperText}
                              </FormHelperText> : null
                            }

                          </FormControl>
                          <Button
                            type="submit"
                            fullWidth
                            variant="contained"
                            color="primary"
                            className={classes.buttonBar}
                          >
                            Sign in
                          </Button>
                        </form>
                      </Paper>
                    </main>
                  </div>
                </div>
              </Grid>
            </Grid>
          </Grid>
        </div>
      </React.Fragment>
    )
  }
}

export default withRouter(withStyles(styles)(LoginForm))
