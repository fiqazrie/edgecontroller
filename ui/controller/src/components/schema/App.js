// SPDX-License-Identifier: Apache-2.0
// Copyright (c) 2019 Intel Corporation

export default {
  schema: {
    type: "object",
    title: "App",
    required: [
      "type",
      "name",
      "version",
      "vendor",
      "cores",
      "memory",
      "source"
    ],
    properties: {
      id: {
        type: "string",
        title: "ID",
        readonly: true
      },
      type: {
        type: "string",
        title: "Type",
        enum: [
          "container",
          "vm"
        ]
      },
      name: {
        type: "string",
        title: "Name"
      },
      version: {
        type: "string",
        title: "Version"
      },
      vendor: {
        type: "string",
        title: "Vendor"
      },
      description: {
        type: "string",
        title: "Description"
      },
      cores: {
        type: "number",
        title: "Cores",
        minimum: 1,
        maximum: 8,
      },
      memory: {
        type: "number",
        title: "Memory (in MB)",
        minimum: 1,
        maximum: 16384,
      },
      ports: {
        type: "array",
        title: "Ports",
        items: {
          type: "object",
          title: "Port",
          properties: {
            port: {
              type: "number",
              title: "Port",
              minimum: 1,
              maximum: 65535,
            },
            protocol: {
              type: "string",
              title: "Protocol",
            },
          },
        },
      },
      source: {
        type: "string",
        title: "Source",
      },
      epafeatures: {
        type: "array",
        title: "EPA Features",
        items: {
          type: "object",
          title: "EPA Feature",
          properties: {
            key: {
              type: "string",
              title: "EPA Feature Key",
            },
            value: {
              type: "string",
              title: "EPA Feature Value",
            },
          },
        },
      }
    }
  },
  form: [
    "*"
  ]
};
