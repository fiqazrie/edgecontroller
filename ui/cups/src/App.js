// SPDX-License-Identifier: Apache-2.0
// Copyright (c) 2019-2020 Intel Corporation

import React, { Component } from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';
import { BrowserRouter as Router } from "react-router-dom";
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Userplanes from './components/Userplanes';
import Userplane from './components/Userplane';
import Header from './components/Header';
import ErrorBoundary from './components/ErrorBoundary';

const useStyles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
  },
  header: {
    padding: 20,
    flexGrow: 1,
  },
  main: {
    width: 'auto',
    marginTop: theme.spacing(14),
    marginBottom: theme.spacing(2),
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(4))]: {
      width: 800,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
});

class App extends Component {
  render() {
    const { classes } = this.props;

    return (
      <Router>
        <div>
          <header>
            <Header className={classes.header} />
          </header>
          <main className={classes.main}>
            <ErrorBoundary>
              <Switch>
                <Route
                  exact
                  path="/"
                  render={() => <Redirect to="/userplanes" />}
                />
                <Route
                  exact
                  path="/userplanes"
                  component={Userplanes}
                />
                <Route
                  exact
                  path="/userplanes/create"
                  render={(props) => <Userplane {...props} createMode={true} />}
                />
                <Route
                  exact
                  path="/userplanes/:id"
                  component={Userplane}
                />
                <Route
                  render={() => <span>404 Not Found</span>}
                />
              </Switch>
            </ErrorBoundary>
          </main>
        </div>
      </Router>
    )
  }
}

App.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(useStyles)(App);
