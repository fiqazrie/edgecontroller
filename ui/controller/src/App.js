// SPDX-License-Identifier: Apache-2.0
// Copyright (c) 2019 Intel Corporation

import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { MuiThemeProvider } from '@material-ui/core/styles';
import './App.css';
import Routes from './routes';
import MuiTheme from './MuiTheme';
import '@material-ui/icons';
import 'typeface-lato';
import 'typeface-roboto';
import { SnackbarProvider } from 'notistack';
import CssBaseline from '@material-ui/core/CssBaseline';
import ErrorBoundary from './components/common/ErrorBoundary';
import OrchestrationContext, {
  initialOrchestrationValue,
} from './context/orchestrationContext';

class App extends Component {
  render() {
    return (
      <React.Fragment>
        <CssBaseline />
        <SnackbarProvider
          maxSnack={3}
          anchorOrigin={{
            vertical: 'top',
            horizontal: 'right',
          }}
        >
          <div>
            <ErrorBoundary>
              <MuiThemeProvider theme={MuiTheme}>
                <BrowserRouter>
                  <OrchestrationContext.Provider
                    value={initialOrchestrationValue}
                  >
                    <Routes />
                  </OrchestrationContext.Provider>
                </BrowserRouter>
              </MuiThemeProvider>
            </ErrorBoundary>
          </div>
        </SnackbarProvider>
      </React.Fragment>
    );
  }
}

export default App;
