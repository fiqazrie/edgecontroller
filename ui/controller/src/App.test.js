// SPDX-License-Identifier: Apache-2.0
// Copyright (c) 2019 Intel Corporation

import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import { mount, shallow } from 'enzyme';
import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import Routes from "./routes";
import MuiTheme from "./MuiTheme";
import Login from "./components/Login";
import renderer from 'react-test-renderer';

describe('Smoke Tests for Main AppView.js', () => {
  beforeEach(() => {
    // values stored in tests will also be available in other tests unless you run
    sessionStorage.clear();
  });

  // Use React Dom to render App Component
  it('renders without crashing (ReactDOM)', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Use Enzyme Shallow rendering for App Component
  it('renders without crashing (Shallow)', () => {
    const sut = shallow(<App />);
    expect(sut.contains('div'));
  });

  // Use Enzyme Mount for App Component
  it('renders without crashing (Mount)', () => {
    const sut = mount(<App />);
    const ThemeWrapper = (
      <MuiThemeProvider theme={MuiTheme}>
        <BrowserRouter>
          <Routes />
        </BrowserRouter>
      </MuiThemeProvider>
    );

    expect(sut.contains(ThemeWrapper)).toEqual(true);
  });

  // Validate Route exists
  it('Prompts Login screen', () => {
    const sut = mount(<App />);

    expect(sut.contains(Login));

    // Expect total of 3 input field
    expect(sut.find('input')).toHaveLength(2);
  });

  it('Validate App Screen snapshot', () => {
    const appTree = renderer.create(<App />).toJSON();

    expect(appTree).toMatchSnapshot();
  });
});
