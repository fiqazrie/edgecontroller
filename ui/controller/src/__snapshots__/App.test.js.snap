// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke Tests for Main AppView.js Validate App Screen snapshot 1`] = `
<div>
  <div>
    <div
      className="LoginForm-root-273"
    >
      <div
        className="MuiGrid-container-285 MuiGrid-justify-xs-center-302"
      >
        <div
          className="MuiGrid-container-285 MuiGrid-spacing-xs-24-309 MuiGrid-align-items-xs-center-293 MuiGrid-justify-xs-center-302 LoginForm-grid-274"
        >
          <div
            className="MuiGrid-item-286 MuiGrid-grid-xs-12-325 MuiGrid-grid-lg-6-361"
          >
            <div
              className="LoginForm-stepContainer-277"
            >
              <div
                className="LoginForm-smallContainer-275"
              >
                <main>
                  <div
                    className="MuiPaper-root-382 MuiPaper-elevation2-386 MuiPaper-rounded-383 LoginForm-paper-281"
                  >
                    <div
                      className="MuiAvatar-root-409 MuiAvatar-colorDefault-410"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-412"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <g
                          fill="none"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                          />
                          <path
                            d="M0 0h24v24H0V0z"
                            opacity=".87"
                          />
                        </g>
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                        />
                      </svg>
                    </div>
                    <h1
                      className="MuiTypography-root-421 MuiTypography-h5-437"
                    >
                      Controller Login
                    </h1>
                    <form
                      autoComplete="off"
                      onSubmit={[Function]}
                    >
                      <div
                        className="MuiFormControl-root-457 MuiFormControl-marginNormal-458 MuiFormControl-fullWidth-460"
                      >
                        <label
                          className="MuiFormLabel-root-472 MuiFormLabel-required-477 MuiInputLabel-required-465 MuiInputLabel-root-461 MuiInputLabel-formControl-466 MuiInputLabel-animated-469"
                          data-shrink={false}
                          htmlFor="username"
                        >
                          Username
                          <span
                            className="MuiFormLabel-asterisk-478"
                          >
                             *
                          </span>
                        </label>
                        <div
                          className="MuiInputBase-root-492 MuiInput-root-479 MuiInput-underline-483 MuiInputBase-formControl-493 MuiInput-formControl-480"
                          onClick={[Function]}
                        >
                          <input
                            aria-invalid={false}
                            autoComplete="off"
                            autoFocus={true}
                            className="MuiInputBase-input-502 MuiInput-input-487"
                            disabled={false}
                            id="username"
                            name="username"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={true}
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        className="MuiFormControl-root-457 MuiFormControl-marginNormal-458 MuiFormControl-fullWidth-460 LoginForm-loginBox-284"
                      >
                        <label
                          className="MuiFormLabel-root-472 MuiFormLabel-required-477 MuiInputLabel-required-465 MuiInputLabel-root-461 MuiInputLabel-formControl-466 MuiInputLabel-animated-469"
                          data-shrink={false}
                          htmlFor="password"
                        >
                          Password
                          <span
                            className="MuiFormLabel-asterisk-478"
                          >
                             *
                          </span>
                        </label>
                        <div
                          className="MuiInputBase-root-492 MuiInput-root-479 MuiInput-underline-483 MuiInputBase-formControl-493 MuiInput-formControl-480"
                          onClick={[Function]}
                        >
                          <input
                            aria-describedby="component-error-text"
                            aria-invalid={false}
                            autoComplete="off"
                            className="MuiInputBase-input-502 MuiInput-input-487 MuiInputBase-inputType-505 MuiInput-inputType-490"
                            disabled={false}
                            id="password"
                            name="password"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={true}
                            type="password"
                          />
                        </div>
                      </div>
                      <button
                        className="MuiButtonBase-root-535 MuiButton-root-509 MuiButton-contained-520 MuiButton-containedPrimary-521 MuiButton-raised-523 MuiButton-raisedPrimary-524 MuiButton-fullWidth-534 LoginForm-buttonBar-279"
                        disabled={false}
                        onBlur={[Function]}
                        onContextMenu={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex="0"
                        type="submit"
                      >
                        <span
                          className="MuiButton-label-510"
                        >
                          Sign in
                        </span>
                        <span
                          className="MuiTouchRipple-root-538"
                        />
                      </button>
                    </form>
                  </div>
                </main>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
